WordPress/gutenberg
# Site Editor: Add template tab to sidebar

## Description
Add template tab to settings sidebar. Resolves https://github.com/WordPress/gutenberg/issues/27470

## How has this been tested?
1. Open site editor
2. Open settings sidebar
3. Make sure there are two tabs: Template and Block
4. Make sure Template tab is selected when opening the settings sidebar with no blocks are selected
5. Make sure Block tab is selected when opening the settings sidebar when a block is selected

## Screenshots 

https://user-images.githubusercontent.com/2256104/106467518-0e731c00-649d-11eb-9adf-25d161eba5ca.mov


## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
